### PR TITLE
Add `no-confusing-arrow` ESLint rule

### PR DESCRIPTION
**Pull request by ntwb**
> Which issue, if any, is this issue related to?

See https://github.com/stylelint/stylelint/pull/4453#issuecomment-570963865

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self explanatory."

The issues detected by the addition of this rule will require updating

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -166,6 +166,7 @@
       "dot-notation": "error",
       "func-name-matching": "error",
       "guard-for-in": "error",
+      "no-confusing-arrow": "error",
```
Comment (hudochenkov) [inline package.json]: Needs to be updated to avoid conflicts with Prettier: https://github.com/prettier/eslint-config-prettier#no-confusing-arrow

Comment (ntwb) [reply package.json]: Ha, thanks, I thought this rule wasn't in the Prettier list 😏

Fixed in https://github.com/stylelint/stylelint/pull/4514/commits/0c7ea8da21bc74530384186ea62075c92e7fa7df

**Comment by XhmikosR** [comment]
Don't forget to add this in the upstream open PR if it gets merged

**Comment by hudochenkov** [comment]
@ntwb would you like to fix lint errors? :)

**Comment by jeddy3** [comment]
Is this ready for review?

**Comment by hudochenkov** [comment]
@jeddy3 yes.

**Comment by XhmikosR** [comment]
@ntwb can you fix this? Otherwise we might as well drop the patch.

**Comment by XhmikosR** [comment]
Ah, wait, it seems @hudochenkov fixed it. Is there anything block from merging? For what is worth, I personally don't see any value in this rule.

**Comment by jeddy3** [comment]
Shall we close this so that we can merge https://github.com/stylelint/stylelint/pull/4453?

**Comment by hudochenkov** [comment]
Why close and not merge?

**Comment by jeddy3** [comment]
Merge works for me too :)

**Pull request merged**